app/scripts/controllers/network/network-controller.js

```diff
@@ -22,14 +23,20 @@ import {
   CHAIN_IDS,
   NETWORK_TYPES,
 } from '../../../../shared/constants/network';
-import {
-  isPrefixedFormattedHexString,
-  isSafeChainId,
-} from '../../../../shared/modules/network.utils';
 import getFetchWithTimeout from '../../../../shared/modules/fetch-with-timeout';
 import createInfuraClient from './createInfuraClient';
 import createJsonRpcClient from './createJsonRpcClient';
 
+/**
+ * @typedef {object} NetworkConfiguration
+ * @property {string} rpcUrl - RPC target URL.
+ * @property {string} chainId - Network ID as per EIP-155
+ * @property {string} chainName - Personalized network name.
+ * @property {string} ticker - Currency ticker.
+ * @property {object} rpcPrefs - Personalized preferences.
+ * @property {string} [networkConfigurationId] - the unique identifier for this networkConfiguration.
```

Comment:
It looks like this property doesn't exist on this type in practice

Reply:
Yep that's correct. Removing.

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/6dbe2ac83e0e2890d7670339979f8bc4e005e4e4